crates/interface-types/src/lib.rs

```diff
@@ -349,48 +321,6 @@ trait TranslateContext {
     unsafe fn get_memory(&mut self) -> Result<&mut [u8]>;
 }
 
-struct RawTranslateContext<'a> {
-    cx: &'a mut Context,
-    handle: &'a mut InstanceHandle,
-}
-
-impl<'a> RawTranslateContext<'a> {
-    fn new(cx: &'a mut Context, handle: &'a mut InstanceHandle) -> Self {
-        RawTranslateContext { cx, handle }
-    }
-}
-
-impl TranslateContext for RawTranslateContext<'_> {
-    fn invoke_alloc(&mut self, alloc_func_name: &str, len: i32) -> Result<i32> {
-        let alloc_args = vec![RuntimeValue::I32(len)];
-        let results = match self.cx.invoke(self.handle, alloc_func_name, &alloc_args)? {
-            ActionOutcome::Returned { values } => values,
-            ActionOutcome::Trapped { message } => bail!("trapped: {}", message),
-        };
-        if results.len() != 1 {
-            bail!("allocator function wrong number of results");
-        }
-        Ok(match results[0] {
-            RuntimeValue::I32(i) => i,
-            _ => bail!("allocator function bad return type"),
-        })
-    }
-    unsafe fn get_memory(&mut self) -> Result<&mut [u8]> {
-        let memory = self
-            .handle
-            .lookup("memory")
-            .ok_or_else(|| format_err!("no exported `memory`"))?;
-        let definition = match memory {
-            wasmtime_runtime::Export::Memory { definition, .. } => definition,
-            _ => bail!("export `memory` wasn't a `Memory`"),
-        };
-        Ok(slice::from_raw_parts_mut(
-            (*definition).base,
-            (*definition).current_length,
-        ))
-    }
-}
-
 struct InstanceTranslateContext(pub api::HostRef<api::Instance>);
 
 impl TranslateContext for InstanceTranslateContext {
```

Comment:
Would it be possible now to remove this trait?

Reply:
Yes, I'll remove that in follow up PR + RuntimeValue dependency.